### PR TITLE
Adding sorting function based on metric to get_job_metrics of nnicli

### PR DESCRIPTION
**Pull request by prismformore**
Adding sorting function (both maximize and minimize) to get_job_metrics of nnicli.
This will be helpful for those who can not access webui.

### CODE REVIEW

**Comment by prismformore** [comment]
fail

File: src/sdk/pycli/nnicli/nni_client.py [hunk 1]
```diff
@@ -129,10 +130,20 @@ def get_job_statistics():
...
 
-def get_job_metrics(trial_job_id=None):
+def get_job_metrics(trial_job_id=None, sort=None):
     """return trial job metrics"""
     api_path = METRICS_PATH if trial_job_id is None else os.path.join(METRICS_PATH, trial_job_id)
-    return _nni_rest_get(api_path)
+    job_metrics = _nni_rest_get(api_path)
+    if sort == 'max' and trial_job_id == None:
+        return sorted(job_metrics, key=cmp_to_key(lambda x, y: -1 \
+                                                  if float(x['data'].replace('"', '')) < float(y['data'].replace('"', '')) \
```
Comment (tabVersion) [inline src/sdk/pycli/nnicli/nni_client.py:139]: Maybe you can use `eval()` to handle serialized data.

**Comment by tabVersion** [comment]
How your code handles dict metric?
Does your code always sort by the default key?

File: src/sdk/pycli/nnicli/nni_client.py [hunk 2]
```diff
@@ -129,10 +130,20 @@ def get_job_statistics():
     """return trial job statistics information as a dict"""
     return _nni_rest_get(JOB_STATISTICS_PATH)
 
-def get_job_metrics(trial_job_id=None):
+def get_job_metrics(trial_job_id=None, sort=None):
```
Comment (chicm-ms) [inline src/sdk/pycli/nnicli/nni_client.py:133]: `nni_client.py` is only a wrapper of rest api, this is not a good place to add additional features to rest api.
Suggest to:
1.  discuss with team whether to add sort feature into the datastore API.
OR
2.  let end user of nnicli to do the sorting.

**Comment by QuanluZhang** [comment]
@prismformore thanks for your contribution. This function is useful. But since it is relatively simple, we believe users can do it by themselves after getting the results from nnictl

**Comment by QuanluZhang** [comment]
@prismformore feel free to reopen if you have more questions about this.

**Pull request closed without merging**